RecurrentFF/benchmarks/Moving-MNIST/moving_mnist.py

```diff
@@ -0,0 +1,272 @@
+
+        y_pos = self.data_chunk['labels'][idx - self.data_chunk_idx]
+        y_neg = y_pos
+        while y_neg == y_pos:
+            y_neg = torch.randint(0, NUM_CLASSES, (1,)).item()
+
+        positive_one_hot_labels = torch.zeros(NUM_CLASSES)
+        positive_one_hot_labels[y_pos] = 1.0
+
+        negative_one_hot_labels = torch.zeros(NUM_CLASSES)
+        negative_one_hot_labels[y_neg] = 1.0
+
+        return (sequences, sequences), (positive_one_hot_labels, negative_one_hot_labels)
+
+    def _load_data_loop(self, data_queue, load_event):
+        """
+        An internal method that runs in a separate process and loads data
+        chunks.
+
+        The method waits for a load event before loading data. It then loads as
+        many chunks as it can into the queue.
+
+        Parameters
+        ----------
+        data_queue : multiprocessing.Manager().Queue()
+            The queue to which data chunks are loaded.
+        load_event : multiprocessing.Manager().Event()
+            An event object used for signaling when new data should be loaded.
+        """
+        while True:
+            load_event.wait()  # Wait for a signal to load data
+
+            # Load chunks into the queue until it is full or all data files have been loaded
+            while not data_queue.full() and self.file_idx < len(self.data_files):
+                chunk_data = torch.load(os.path.join(
+                    self.root_dir, self.data_files[self.file_idx]))
+                data_queue.put(chunk_data)
+                self.file_idx += 1
+
+            load_event.clear()  # Clear the event after data is loaded
+
+            # reset the loop
+            if self.file_idx >= len(self.data_files):
+                self.file_idx = 0
+
+
+def train_collate_fn(batch):
+    data_batch, label_batch = zip(*batch)
+
+    data1, data2 = zip(*data_batch)
+    positive_labels, negative_labels = zip(*label_batch)
+
+    data1 = torch.stack(data1, 1)
+    data2 = torch.stack(data2, 1)
+    positive_labels = torch.stack(positive_labels)
+    negative_labels = torch.stack(negative_labels)
+
+    positive_labels = positive_labels.unsqueeze(0).repeat(ITERATIONS, 1, 1)
+    negative_labels = negative_labels.unsqueeze(0).repeat(ITERATIONS, 1, 1)
+
+    return TrainInputData(data1, data2), TrainLabelData(positive_labels, negative_labels)
+
+
+def test_collate_fn(batch):
+    data_batch, label_batch = zip(*batch)
+
+    pos_data, _neg_data = zip(*data_batch)
+    positive_labels, _negative_labels = zip(*label_batch)
+
+    pos_data = torch.stack(pos_data, 1)
+    positive_labels = torch.stack(positive_labels)
+
+    positive_labels = positive_labels.argmax(dim=1)
+
+    return TestData(pos_data, positive_labels)
+
+
+def MNIST_loaders(train_batch_size, test_batch_size):
+    # TODO: need a transform? Similar to MNIST:
+    # transform = Compose([
+    #     ToTensor(),
+    #     Normalize((0.1307,), (0.3081,)),
+    #     Lambda(lambda x: torch.flatten(x))])
```

Comment:
Should we normalize? I am doing so for MNIST, but not here. 

Maybe the model is sensitive to one format over another and it is worth it to see.